<div style="
    background-color: white;
    color: black;
    padding: 15px;
    text-align: center;
    font-family: Georgia, serif;
">

<h1 style="
    color: black;
    font-size: 32px;
    margin: 0;
">
ANALYSIS OF CUSTOMER BEHAVIOR IN 
         E-COMMERCE

</h1>

</div>

<div style="text-align: center;">

<img src="customer.jpg" 
     style="width: 100%; max-width: 1200px; border-radius: 10px;">

</div>

<h3 style="
  color: #FFFFFF;
  background-color: #003366;
  padding: 6px 14px;
  text-align: center;
  border-radius: 6px;
  font-family: 'Georgia', serif;
  font-weight: 500;
  font-size: 25px;
  letter-spacing: 1px;
">
  INTRODUCTION
</h3>

</div>
<p style="
    font-family: Arial, sans-serif;
    font-size: 17px;
    color: black;
    line-height: 1.7;
    text-align: justify;
">


<p style="
  font-family: Arial, sans-serif;
  font-size: 17px;
  line-height: 1.7;
  text-align: justify;
">
E-commerce has changed the way customers search for, compare, and purchase products. Understanding customer behaviour is important for identifying purchasing patterns, customer preferences, and factors that influence online shopping decisions. This project, “Analysis of Customer Behavior in E-Commerce,” analyses customer information and shopping activities using Python and Pandas. The study examines customer demographics, device usage, traffic sources, browsing activity, product information, cart behaviour, purchases, payment methods, and preferred shopping time. Basic data analysis techniques such as missing-value treatment, duplicate checking, filtering, sorting, frequency analysis, and statistical calculations are used to understand the dataset.

</p>

<h3 style="
  color: #FFFFFF;
  background-color: #003366;
  padding: 6px 14px;
  text-align: center;
  border-radius: 6px;
  font-family: 'Georgia', serif;
  font-weight: 500;
  font-size: 25px;
  letter-spacing: 1px;
">
  Abstract
</h3>

</div>
<p style="
    font-family: Arial, sans-serif;
    font-size: 17px;
    color: black;
    line-height: 1.7;
    text-align: justify;
">


<p style="
  font-family: Arial, sans-serif;
  font-size: 17px;
  line-height: 1.7;
  text-align: justify;
">
This project analyses an e-commerce customer behaviour dataset containing information about customers and their online shopping activities. The dataset includes demographic, session, product, cart, purchase, payment, and time-related variables. Using Pandas in Jupyter Notebook, the data is explored through rows, columns, data types, statistical summaries, and missing-value analysis. Missing values are treated using appropriate methods, including mean values for numerical columns and “Not Available” for selected categorical columns. The project further examines customer age, gender, product categories, abandoned carts, purchases, discounts, payment methods, shopping time, locations, and product prices. The analysis provides a basic understanding of customer behaviour in an e-commerce environment.

</p>

<h3 style="
  color: #FFFFFF;
  background-color: #003366;
  padding: 6px 14px;
  text-align: center;
  border-radius: 6px;
  font-family: 'Georgia', serif;
  font-weight: 500;
  font-size: 25px;
  letter-spacing: 1px;
">
  About the dataset
</h3>


</div>
<p style="
    font-family: Arial, sans-serif;
    font-size: 17px;
    color: black;
    line-height: 1.7;
    text-align: justify;
">


<p style="
  font-family: Arial, sans-serif;
  font-size: 17px;
  line-height: 1.7;
  text-align: justify;
">
The dataset is an E-Commerce Customer Behaviour dataset used to analyse customer shopping patterns. It contains information related to individual customers and their shopping sessions. The dataset covers variables such as customer ID, session ID, age, gender, location, device type, traffic source, session duration, pages viewed, product category, product price, discount, cart activity, purchase status, payment method, and time of day.
In your notebook, the dataset is loaded from ecommerce_customer_behavior.csv using Pandas. The data is examined for its number of rows and columns, data types, missing values, and duplicate records. Missing values are filled for selected columns before further analysis.

</p>

<div style="
    width: 92%;
    margin: 35px auto;
    padding: 30px;
    border: 1px solid #555;
    border-radius: 8px;
    background-color: #ffffff !important;
    color: #000000 !important;
">

<h2 style="
    text-align: center;
    margin-top: 0;
    margin-bottom: 25px;
    font-size: 25px;
    letter-spacing: 1px;
    color: #000000 !important;
">
Column Description
</h2>

<table style="
    width: 100%;
    border-collapse: collapse;
    font-size: 14px;
    background-color: #ffffff !important;
    color: #000000 !important;
">

<tr style="
    background-color: #ffffff !important;
    color: #000000 !important;
">
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Column Name</th>
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Raw Type</th>
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Target Type</th>
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Description</th>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">customer_id</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Unique identification number assigned to each customer.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">session_id</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Unique identification number assigned to each shopping session.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">age</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Age of the customer.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">gender</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Gender of the customer.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">location</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Location or city of the customer.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">device_type</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Type of device used by the customer for online shopping.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">traffic_source</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Source through which the customer reached the e-commerce platform.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">session_duration_minutes</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Duration of the customer's shopping session in minutes.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">pages_viewed</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Number of pages viewed during the shopping session.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">product_category</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Category of the product viewed or purchased.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">product_price</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Price of the product involved in the shopping session.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">discount_applied</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Indicates whether a discount was applied to the product.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">add_to_cart</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Indicates whether the customer added a product to the shopping cart.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">abandoned_cart</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Indicates whether the customer abandoned the shopping cart.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">purchase_made</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Indicates whether the customer completed a purchase.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">payment_method</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Payment method used by the customer.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">time_of_day</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Time period during which the shopping session took place.</td>
</tr>

</table>
</div>

In [1]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [2]:
#load dataset from CSV file
df=pd.read_csv("ecommerce_customer_behavior.csv")

In [3]:
#display first 5 row
df.head()

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
0,CUST00001,SESS784673,39.0,Male,Singapore,Mobile,Organic,1.9,1.0,Fashion,132.20,No,Yes,No,Yes,Wallet,Afternoon
1,CUST00002,SESS647875,63.0,Female,London,Desktop,Organic,22.9,16.0,Home & Kitchen,457.67,No,Yes,No,Yes,Cash on Delivery,Evening
2,CUST00003,SESS277337,24.0,Male,Phoenix,Desktop,Social,4.7,3.0,Grocery,62.18,No,Yes,No,Yes,Credit Card,Evening
3,CUST00004,SESS791368,30.0,Female,Houston,Mobile,Social,1.9,1.0,Electronics,648.55,No,No,No,No,NaN,Morning
4,CUST00005,SESS339305,64.0,Male,Los Angeles,Desktop,Organic,1.0,1.0,Beauty,123.66,No,Yes,Yes,No,NaN,Night


In [4]:
#display last 5 row
df.tail()

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
1595,CUST01596,SESS681874,62.0,Female,Houston,Mobile,Organic,4.9,6.0,Home & Kitchen,38.74,No,No,No,No,NaN,Night
1596,CUST01597,SESS827775,25.0,Male,London,Desktop,Organic,7.4,7.0,Toys,80.89,Yes,Yes,No,No,NaN,Night
1597,CUST01598,SESS413011,41.0,Male,Singapore,Mobile,Social,16.0,NaN,Electronics,168.47,No,No,No,No,NaN,Evening
1598,CUST01599,SESS600864,21.0,Male,London,Desktop,Social,3.0,1.0,Sports,155.45,Yes,Yes,No,Yes,Credit Card,Afternoon
1599,CUST01600,SESS245709,25.0,Male,Singapore,Mobile,Organic,4.4,1.0,Beauty,139.24,No,No,No,No,NaN,Night


In [5]:
#show number of rows and columns
df.shape

(1600, 17)

In [6]:
df.size

27200

In [7]:
#display column names
df.columns

Index(['customer_id', 'session_id', 'age', 'gender', 'location', 'device_type',
       'traffic_source', 'session_duration_minutes', 'pages_viewed',
       'product_category', 'product_price', 'discount_applied', 'add_to_cart',
       'abandoned_cart', 'purchase_made', 'payment_method', 'time_of_day'],
      dtype='object')

In [8]:
#show datatypes of all columns
df.dtypes

customer_id                  object
session_id                   object
age                         float64
gender                       object
location                     object
device_type                  object
traffic_source               object
session_duration_minutes    float64
pages_viewed                float64
product_category             object
product_price               float64
discount_applied             object
add_to_cart                  object
abandoned_cart               object
purchase_made                object
payment_method               object
time_of_day                  object
dtype: object

In [9]:
#display complete information about dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1600 entries, 0 to 1599
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   customer_id               1600 non-null   object 
 1   session_id                1600 non-null   object 
 2   age                       1543 non-null   float64
 3   gender                    1600 non-null   object 
 4   location                  1600 non-null   object 
 5   device_type               1600 non-null   object 
 6   traffic_source            1517 non-null   object 
 7   session_duration_minutes  1541 non-null   float64
 8   pages_viewed              1522 non-null   float64
 9   product_category          1600 non-null   object 
 10  product_price             1529 non-null   float64
 11  discount_applied          1600 non-null   object 
 12  add_to_cart               1600 non-null   object 
 13  abandoned_cart            1600 non-null   object 
 14  purchase

### Observation:
All 1,600 records and 17 columns are identified, with four numerical variables and thirteen categorical variables.

In [10]:
#statistical summery of numerical columns
df.describe()

,age,session_duration_minutes,pages_viewed,product_price
count,1543.000000,1541.000000,1522.000000,1529.000000
mean,41.288399,7.834004,5.145204,226.458679
std,13.753779,7.376551,4.620717,250.585495
min,18.000000,1.000000,1.000000,5.020000
25%,29.000000,2.300000,1.000000,56.490000
50%,41.000000,5.400000,4.000000,142.520000
75%,53.000000,10.900000,7.000000,282.530000
max,64.000000,56.300000,37.000000,1199.660000


### Observation:
The code references the describe method but does not execute it, because () is missing.

In [50]:
#check missing values
df.isnull().sum()

customer_id                 0
session_id                  0
age                         0
gender                      0
location                    0
device_type                 0
traffic_source              0
session_duration_minutes    0
pages_viewed                0
product_category            0
product_price               0
discount_applied            0
add_to_cart                 0
abandoned_cart              0
purchase_made               0
payment_method              0
time_of_day                 0
dtype: int64

### Observation:
Missing values are present in several columns, especially payment_method (1,223)

In [51]:
#total missing values in dataset
df.isnull().sum().sum()

np.int64(0)

### Observation:
The dataset initially contains 1,521 missing values in total.

# <center>FILL MISSING VALUES

In [13]:
#fill missing age with mean age
df["age"].fillna(df["age"].mean(),inplace=True)

### Observation:
Missing age values are replaced with the mean age.

In [14]:
df["traffic_source"].fillna("Not Available",inplace=True)

### Observation:
Missing traffic-source values are replaced with “Not Available”.

In [15]:
df["session_duration_minutes"].fillna(df["session_duration_minutes"].mean(),inplace=True)

### Observation:
Missing session-duration values are replaced with the mean session duration.

In [16]:
df["pages_viewed"].fillna(df["pages_viewed"].mean(),inplace=True)

### Observation:
Missing pages-viewed values are replaced with the mean number of pages viewed.

In [17]:
df["product_price"].fillna(df["product_price"].mean(),inplace=True)

### Observation:
Missing product-price values are replaced with the mean product price.

In [18]:
df["payment_method"].fillna("Not Available",inplace=True)

### Observation:
Missing payment-method values are replaced with “Not Available”.

In [19]:
#recheking if any null values exist
df.isnull().sum()

customer_id                 0
session_id                  0
age                         0
gender                      0
location                    0
device_type                 0
traffic_source              0
session_duration_minutes    0
pages_viewed                0
product_category            0
product_price               0
discount_applied            0
add_to_cart                 0
abandoned_cart              0
purchase_made               0
payment_method              0
time_of_day                 0
dtype: int64

### Observation:
After treatment, all columns have zero missing values

In [20]:
#check for count of duplicate values
df.duplicated().sum()

np.int64(0)

### Observation:
The dataset contains 0 duplicate records.

In [21]:
#display specific column
df["device_type"]

0        Mobile
1       Desktop
2       Desktop
3        Mobile
4       Desktop
         ...   
1595     Mobile
1596    Desktop
1597     Mobile
1598    Desktop
1599     Mobile
Name: device_type, Length: 1600, dtype: object

In [22]:
#display multiple columns
df[["device_type","age","gender"]]

,device_type,age,gender
0,Mobile,39.0,Male
1,Desktop,63.0,Female
2,Desktop,24.0,Male
3,Mobile,30.0,Female
4,Desktop,64.0,Male
...,...,...,...
1595,Mobile,62.0,Female
1596,Desktop,25.0,Male
1597,Mobile,41.0,Male
1598,Desktop,21.0,Male


In [23]:
#display first customer record
df.iloc[0]

customer_id                  CUST00001
session_id                  SESS784673
age                               39.0
gender                            Male
location                     Singapore
device_type                     Mobile
traffic_source                 Organic
session_duration_minutes           1.9
pages_viewed                       1.0
product_category               Fashion
product_price                    132.2
discount_applied                    No
add_to_cart                        Yes
abandoned_cart                      No
purchase_made                      Yes
payment_method                  Wallet
time_of_day                  Afternoon
Name: 0, dtype: object

In [24]:
#display first five row
df.iloc[0:5]

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
0,CUST00001,SESS784673,39.0,Male,Singapore,Mobile,Organic,1.9,1.0,Fashion,132.20,No,Yes,No,Yes,Wallet,Afternoon
1,CUST00002,SESS647875,63.0,Female,London,Desktop,Organic,22.9,16.0,Home & Kitchen,457.67,No,Yes,No,Yes,Cash on Delivery,Evening
2,CUST00003,SESS277337,24.0,Male,Phoenix,Desktop,Social,4.7,3.0,Grocery,62.18,No,Yes,No,Yes,Credit Card,Evening
3,CUST00004,SESS791368,30.0,Female,Houston,Mobile,Social,1.9,1.0,Electronics,648.55,No,No,No,No,Not Available,Morning
4,CUST00005,SESS339305,64.0,Male,Los Angeles,Desktop,Organic,1.0,1.0,Beauty,123.66,No,Yes,Yes,No,Not Available,Night


In [25]:
#display row 0 and columns 1 to 3
df.iloc[0,1:4]

session_id    SESS784673
age                 39.0
gender              Male
Name: 0, dtype: object

In [26]:
#display customers older than 30
df[df["age"]>30]

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
0,CUST00001,SESS784673,39.0,Male,Singapore,Mobile,Organic,1.9,1.000000,Fashion,132.20,No,Yes,No,Yes,Wallet,Afternoon
1,CUST00002,SESS647875,63.0,Female,London,Desktop,Organic,22.9,16.000000,Home & Kitchen,457.67,No,Yes,No,Yes,Cash on Delivery,Evening
4,CUST00005,SESS339305,64.0,Male,Los Angeles,Desktop,Organic,1.0,1.000000,Beauty,123.66,No,Yes,Yes,No,Not Available,Night
5,CUST00006,SESS855432,34.0,Female,New York,Desktop,Social,4.5,3.000000,Electronics,384.90,Yes,Yes,Yes,No,Not Available,Afternoon
6,CUST00007,SESS943474,56.0,Female,Toronto,Desktop,Referral,6.6,2.000000,Electronics,324.40,No,No,No,No,Not Available,Evening
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1592,CUST01593,SESS326540,39.0,Male,Phoenix,Mobile,Referral,9.8,10.000000,Books,41.23,No,Yes,No,Yes,Credit Card,Evening
1593,CUST01594,SESS558966,39.0,Female,Houston,Mobile,Social,4.0,3.000000,Electronics,561.90,No,No,No,No,Not Available,Night
1594,CUST01595,SESS479431,61.0,Male,London,Mobile,Organic,13.5,7.000000,Electronics,531.39,Yes,No,No,No,Not Available,Afternoon
1595,CUST01596,SESS681874,62.0,Female,Houston,Mobile,Organic,4.9,6.000000,Home & Kitchen,38.74,No,No,No,No,Not Available,Night


### Observation:
Customers older than 30 years are successfully filtered from the dataset.

In [27]:
#display female customers
df[df["gender"]=="Female"]

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
1,CUST00002,SESS647875,63.0,Female,London,Desktop,Organic,22.9,16.0,Home & Kitchen,457.67,No,Yes,No,Yes,Cash on Delivery,Evening
3,CUST00004,SESS791368,30.0,Female,Houston,Mobile,Social,1.9,1.0,Electronics,648.55,No,No,No,No,Not Available,Morning
5,CUST00006,SESS855432,34.0,Female,New York,Desktop,Social,4.5,3.0,Electronics,384.90,Yes,Yes,Yes,No,Not Available,Afternoon
6,CUST00007,SESS943474,56.0,Female,Toronto,Desktop,Referral,6.6,2.0,Electronics,324.40,No,No,No,No,Not Available,Evening
8,CUST00009,SESS910181,53.0,Female,Sydney,Mobile,Social,19.7,16.0,Books,55.36,Yes,No,No,No,Not Available,Afternoon
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1586,CUST01587,SESS763975,33.0,Female,London,Mobile,Organic,13.6,9.0,Fashion,299.25,No,No,No,No,Not Available,Afternoon
1587,CUST01588,SESS326852,60.0,Female,Phoenix,Mobile,Social,6.6,2.0,Electronics,417.38,No,No,No,No,Not Available,Evening
1589,CUST01590,SESS460900,59.0,Female,Toronto,Mobile,Social,11.5,5.0,Beauty,62.77,No,Yes,No,Yes,Cash on Delivery,Night
1593,CUST01594,SESS558966,39.0,Female,Houston,Mobile,Social,4.0,3.0,Electronics,561.90,No,No,No,No,Not Available,Night


In [28]:
#count female customers
df["gender"].value_counts()

gender
Female    777
Male      770
Other      53
Name: count, dtype: int64

### Observation:
Female customer records are successfully filtered.
There are 777 Female, 770 Male and 53 Other customers.

In [29]:
#count which product categories mainly purchased
df["product_category"].value_counts()

product_category
Fashion           346
Electronics       293
Home & Kitchen    223
Beauty            172
Sports            152
Books             151
Grocery           144
Toys              119
Name: count, dtype: int64

### Observation:
Fashion (346) is the most frequent product category, followed by Electronics (293).

In [30]:
#count without buying it or not
df["abandoned_cart"].value_counts()

abandoned_cart
No     1244
Yes     356
Name: count, dtype: int64

### Observation:
1,244 customers did not abandon their carts, while 356 abandoned their carts.


In [31]:
#count if purchase happend or not
df["purchase_made"].value_counts()

purchase_made
No     1223
Yes     377
Name: count, dtype: int64

### Observation:
1,223 sessions did not result in a purchase, while 377 resulted in a purchase.

In [32]:
#count if discount applied or not
df["discount_applied"].value_counts()

discount_applied
No     999
Yes    601
Name: count, dtype: int64

### Observation:
Discounts were applied in 601 cases and not applied in 999 cases.

In [33]:
#mainly used payment method
df["payment_method"].value_counts()

payment_method
Not Available       1223
Credit Card          136
Cash on Delivery      95
Debit Card            91
Wallet                55
Name: count, dtype: int64

### Observation:
After missing-value treatment, Not Available (1,223) is the largest category; Credit Card has 136 records.

In [34]:
#generaly customer prefering shopping time
df["time_of_day"].value_counts()

time_of_day
Evening      518
Afternoon    451
Night        337
Morning      294
Name: count, dtype: int64

### Observation:
Evening (518) is the most common shopping period, followed by Afternoon (451).

In [35]:
#calculate average age
df["age"].mean()

np.float64(41.28839922229423)

### Observation:
The average customer age is approximately 41.29 years.

In [36]:
#find maximum age
df["age"].max()

64.0

### Observation:
The maximum customer age is 64 years.

In [37]:
#finf minimum age
df["age"].min()

18.0

### Observation:
The minimum customer age is 18 years.

In [38]:
#calculate median age
df["age"].median()

41.28839922229423

### Observation:
The median customer age is approximately 41.29 years.

In [39]:
#calculate sum of product price
df["product_price"].sum()

np.float64(362333.8862001308)

### Observation:
The total product-price value is approximately 362,333.89.

In [40]:
#calculate average product price
df["product_price"].mean()

np.float64(226.45867887508174)

### Observation:
he average product price is approximately 226.46.

In [41]:
#find oldest customer
df[df["age"]==df["age"].max()]

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
4,CUST00005,SESS339305,64.0,Male,Los Angeles,Desktop,Organic,1.0,1.000000,Beauty,123.660000,No,Yes,Yes,No,Not Available,Night
25,CUST00026,SESS917657,64.0,Female,New York,Mobile,Referral,3.8,5.000000,Books,47.930000,No,Yes,No,No,Not Available,Morning
93,CUST00094,SESS839373,64.0,Female,Phoenix,Mobile,Organic,10.9,5.000000,Fashion,196.410000,Yes,No,No,No,Not Available,Evening
114,CUST00115,SESS780877,64.0,Male,Sydney,Mobile,Organic,12.9,10.000000,Fashion,73.270000,Yes,No,No,No,Not Available,Afternoon
131,CUST00132,SESS281609,64.0,Male,Singapore,Tablet,Social,5.3,2.000000,Home & Kitchen,193.080000,No,No,No,No,Not Available,Night
258,CUST00259,SESS618989,64.0,Male,Chicago,Desktop,Ads,10.5,6.000000,Books,33.160000,No,Yes,No,Yes,Debit Card,Afternoon
337,CUST00338,SESS281035,64.0,Female,New York,Desktop,Organic,3.5,5.145204,Beauty,137.670000,No,Yes,No,Yes,Credit Card,Evening
341,CUST00342,SESS950663,64.0,Female,Houston,Tablet,Ads,13.0,12.000000,Fashion,224.380000,No,Yes,Yes,No,Not Available,Afternoon
365,CUST00366,SESS189612,64.0,Male,Los Angeles,Mobile,Not Available,1.0,3.000000,Electronics,563.600000,No,No,No,No,Not Available,Morning
473,CUST00474,SESS577594,64.0,Other,Chicago,Tablet,Ads,12.1,9.000000,Home & Kitchen,339.040000,Yes,Yes,Yes,No,Not Available,Evening


### Observation:
The oldest recorded customer age is 64 years.

In [42]:
#find smallest customer
df[df["age"]==df["age"].min()]

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
44,CUST00045,SESS261690,18.0,Female,London,Desktop,Social,1.300000,2.000000,Fashion,226.458679,Yes,No,No,No,Not Available,Night
51,CUST00052,SESS502251,18.0,Female,London,Tablet,Ads,10.900000,9.000000,Books,55.250000,No,Yes,Yes,No,Not Available,Night
100,CUST00101,SESS604587,18.0,Female,Sydney,Mobile,Organic,2.600000,2.000000,Fashion,294.120000,No,Yes,Yes,No,Not Available,Night
119,CUST00120,SESS326103,18.0,Female,New York,Mobile,Social,3.000000,1.000000,Electronics,906.480000,No,Yes,Yes,No,Not Available,Morning
220,CUST00221,SESS959799,18.0,Male,Los Angeles,Mobile,Social,4.900000,5.145204,Home & Kitchen,98.120000,No,No,No,No,Not Available,Evening
347,CUST00348,SESS811711,18.0,Female,New York,Desktop,Organic,10.100000,7.000000,Sports,33.260000,No,No,No,No,Not Available,Afternoon
367,CUST00368,SESS781345,18.0,Male,Sydney,Desktop,Social,25.300000,17.000000,Toys,21.240000,No,No,No,No,Not Available,Night
431,CUST00432,SESS118440,18.0,Male,Sydney,Mobile,Social,1.000000,1.000000,Home & Kitchen,247.170000,No,No,No,No,Not Available,Afternoon
440,CUST00441,SESS471990,18.0,Female,Toronto,Tablet,Ads,14.500000,9.000000,Electronics,213.180000,No,Yes,Yes,No,Not Available,Night
472,CUST00473,SESS878750,18.0,Male,Los Angeles,Tablet,Ads,1.100000,1.000000,Fashion,158.940000,Yes,Yes,Yes,No,Not Available,Morning


### Observation:
The youngest recorded customer age is 18 years.

In [43]:
#sort by age ascending
df.sort_values("age")

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
1321,CUST01322,SESS331376,18.0,Male,New York,Desktop,Ads,6.100000,6.000000,Electronics,1048.010000,No,No,No,No,Not Available,Evening
220,CUST00221,SESS959799,18.0,Male,Los Angeles,Mobile,Social,4.900000,5.145204,Home & Kitchen,98.120000,No,No,No,No,Not Available,Evening
1563,CUST01564,SESS305465,18.0,Male,New York,Desktop,Organic,5.900000,5.145204,Books,9.840000,No,Yes,No,Yes,Cash on Delivery,Night
1163,CUST01164,SESS736632,18.0,Female,Chicago,Mobile,Organic,8.800000,6.000000,Beauty,125.700000,Yes,Yes,Yes,No,Not Available,Afternoon
1040,CUST01041,SESS731377,18.0,Male,Los Angeles,Mobile,Organic,7.834004,1.000000,Fashion,274.130000,No,No,No,No,Not Available,Evening
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
579,CUST00580,SESS656008,64.0,Female,Los Angeles,Mobile,Organic,13.300000,8.000000,Fashion,226.458679,No,Yes,No,Yes,Credit Card,Night
479,CUST00480,SESS849128,64.0,Male,Singapore,Mobile,Not Available,15.200000,10.000000,Toys,24.620000,No,Yes,No,Yes,Cash on Delivery,Afternoon
337,CUST00338,SESS281035,64.0,Female,New York,Desktop,Organic,3.500000,5.145204,Beauty,137.670000,No,Yes,No,Yes,Credit Card,Evening
25,CUST00026,SESS917657,64.0,Female,New York,Mobile,Referral,3.800000,5.000000,Books,47.930000,No,Yes,No,No,Not Available,Morning


In [44]:
#sort by age descending
df.sort_values("age",ascending=False)

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
774,CUST00775,SESS958711,64.0,Female,Toronto,Desktop,Social,1.600000,1.000000,Grocery,24.12,No,Yes,Yes,No,Not Available,Evening
473,CUST00474,SESS577594,64.0,Other,Chicago,Tablet,Ads,12.100000,9.000000,Home & Kitchen,339.04,Yes,Yes,Yes,No,Not Available,Evening
479,CUST00480,SESS849128,64.0,Male,Singapore,Mobile,Not Available,15.200000,10.000000,Toys,24.62,No,Yes,No,Yes,Cash on Delivery,Afternoon
1206,CUST01207,SESS830804,64.0,Female,New York,Mobile,Social,1.100000,2.000000,Beauty,130.58,No,Yes,Yes,No,Not Available,Afternoon
930,CUST00931,SESS166934,64.0,Male,Toronto,Desktop,Ads,19.400000,10.000000,Grocery,78.61,No,Yes,No,Yes,Cash on Delivery,Afternoon
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
472,CUST00473,SESS878750,18.0,Male,Los Angeles,Tablet,Ads,1.100000,1.000000,Fashion,158.94,Yes,Yes,Yes,No,Not Available,Morning
1563,CUST01564,SESS305465,18.0,Male,New York,Desktop,Organic,5.900000,5.145204,Books,9.84,No,Yes,No,Yes,Cash on Delivery,Night
1040,CUST01041,SESS731377,18.0,Male,Los Angeles,Mobile,Organic,7.834004,1.000000,Fashion,274.13,No,No,No,No,Not Available,Evening
1321,CUST01322,SESS331376,18.0,Male,New York,Desktop,Ads,6.100000,6.000000,Electronics,1048.01,No,No,No,No,Not Available,Evening


In [45]:
#sort by product price descending
df.sort_values("product_price",ascending=False)

,customer_id,session_id,age,gender,location,device_type,traffic_source,session_duration_minutes,pages_viewed,product_category,product_price,discount_applied,add_to_cart,abandoned_cart,purchase_made,payment_method,time_of_day
1218,CUST01219,SESS954075,26.000000,Female,Los Angeles,Desktop,Social,13.3,9.000000,Electronics,1199.66,No,No,No,No,Not Available,Evening
1513,CUST01514,SESS881925,19.000000,Male,New York,Mobile,Referral,2.8,1.000000,Electronics,1196.95,No,Yes,No,Yes,Wallet,Night
752,CUST00753,SESS184119,38.000000,Female,Singapore,Tablet,Referral,18.1,10.000000,Electronics,1188.24,Yes,Yes,No,Yes,Wallet,Afternoon
1270,CUST01271,SESS569422,27.000000,Male,Houston,Mobile,Ads,2.5,1.000000,Electronics,1187.85,No,No,No,No,Not Available,Evening
1508,CUST01509,SESS940668,58.000000,Female,Los Angeles,Mobile,Social,13.6,7.000000,Electronics,1177.01,No,No,No,No,Not Available,Afternoon
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
73,CUST00074,SESS628498,41.288399,Female,Los Angeles,Desktop,Social,1.9,1.000000,Grocery,5.57,No,No,No,No,Not Available,Morning
456,CUST00457,SESS637197,34.000000,Male,New York,Mobile,Ads,13.7,6.000000,Grocery,5.29,No,Yes,Yes,No,Not Available,Night
928,CUST00929,SESS129165,57.000000,Female,New York,Mobile,Social,8.5,3.000000,Books,5.26,Yes,Yes,No,Yes,Credit Card,Evening
1193,CUST01194,SESS173129,41.288399,Female,Singapore,Desktop,Ads,5.9,5.145204,Grocery,5.11,No,No,No,No,Not Available,Afternoon


In [46]:
#average age by gender
df.groupby("gender")["age"].mean()

gender
Female    41.977104
Male      40.943704
Other     36.199562
Name: age, dtype: float64

### Observation:
Average age is approximately 41.98 Female, 40.94 Male and 36.20 Other.

In [47]:
#display unique location values
df["location"].unique()

array(['Singapore', 'London', 'Phoenix', 'Houston', 'Los Angeles',
       'New York', 'Toronto', 'Sydney', 'Dubai', 'Chicago'], dtype=object)

### Observation:
New York (219) has the highest number of customer records, followed by Los Angeles (212) and London (196).

In [48]:
#count unique location values
df["location"].value_counts()

location
New York       219
Los Angeles    212
London         196
Chicago        159
Toronto        157
Houston        153
Sydney         138
Phoenix        128
Dubai          124
Singapore      114
Name: count, dtype: int64

In [49]:
#save modified dataset
df.to_csv("new_ecommerce_customer_behavior.csv",index=False)

<h3 style="
  color: #FFFFFF;
  background-color: #003366;
  padding: 6px 14px;
  text-align: center;
  border-radius: 6px;
  font-family: 'Georgia', serif;
  font-weight: 500;
  font-size: 25px;
  letter-spacing: 1px;
">
  CONCLUSION
</h3>

</div>
<p style="
    font-family: Arial, sans-serif;
    font-size: 17px;
    color: black;
    line-height: 1.7;
    text-align: justify;
">

<p style="
  font-family: Arial, sans-serif;
  font-size: 17px;
  line-height: 1.7;
  text-align: justify;
">
The study “Analysis of Customer Behaviour in E-Commerce” provides a clear understanding of customer characteristics and online shopping patterns. The analysis covered age, gender, product categories, payment methods, discounts, cart abandonment, purchasing behaviour, and shopping time. The dataset initially contained missing values, which were successfully treated, and no duplicate records were found. The findings show different patterns in customer preferences and purchasing activities. Overall, the study helps in understanding e-commerce customer behaviour and can support better decisions in online business and marketing strategies.


</p>